website/notifications/utils.py

```diff
@@ -69,6 +72,12 @@ def remove_contributor_from_subscriptions(node, user):
 
 @signals.node_deleted.connect
 def remove_subscription(node):
+    remove_subscription_task(node._id)
+
+@run_postcommit(once_per_request=False, celery=True)
+@app.task(max_retries=5, default_retry_delay=60)
```

Comment:
Because it was causing transactions to lock and fail on bulk api requests. The main point of this PR is switching three functions to run as celery tasks to work around the locking issue. I would switch the entirety of the postcommit handler to do that but the varnish purging tasks need to run after the transaction is committed but prior to the response being sent to the user so they have to run directly after the database commit.